register.lic

```diff
@@ -0,0 +1,76 @@
+=begin
+  Search a deed register for the specified crafting material.
+  Output all lines that match.
+=end
+
+custom_require.call(%w[common common-items drinfomon])
+
+class Register
+  include DRC
+  include DRCI
+
+  def initialize
+    settings = get_settings
+	@search_results = []
+
+    arg_definitions =
+      [
+        [
+          { name: 'query', regex: /^[A-z\s\-\']+$/i, description: 'crafting material to search for, use quotes for multiple words' }
+        ]
+      ]
+
+    args = parse_args(arg_definitions)
+
+    if args.query
+      stow_hands
+	  if get_item('deed register', get_settings.crafting_container)
+	    fput("turn my register to contents")
+        search(args.query)
+	  else
+	    echo "Can\'t find your deed register."
+		exit
+	  end
+      stow_hands
+	end
+	respond('   ' + "Results:")
+	if @search_results.any?
+	  display_results(@search_results)
+	else
+	  respond('   ' + "No matches found.")
+	end
+  end
+
+  def display_results(results)
+    results.each do |data|
+      respond("#{data}")
```

Comment:
Does string interpolation do anything here?

Reply:
Apparently not.  When I was first writing the script, I had issues with it not finding a line if it had the query in the notes you can write in yourself.

IE: searching for 3.68 would not match
    84 -- a deed for a steel ingot (90V 99Q - 3.68 density MCS --Binu)

It's been a few weeks and now I don't remember which change I made fixed that issue.  I'll test without the interpolation and if it's not necessary I'll remove it.